src/Compilers/Test/Utilities/VisualBasic/CompilationTestUtils.vb

```diff
@@ -879,62 +880,64 @@ Friend Module CompilationUtils
-
-                messages.AppendLine("MISSING ERROR MESSAGES:")
-                For Each l In expectedLines
-                    If Not actualLines.Contains(l) Then
-                        messages.AppendLine(l)
-                        appendedLines += 1
-                    End If
-                Next
-
-                messages.AppendLine("UNEXPECTED ERROR MESSAGES:")
-                For Each l In actualLines
-                    If Not expectedLines.Contains(l) Then
-                        messages.AppendLine(l)
-                        appendedLines += 1
-                    End If
-                Next
-
-                If appendedLines > 0 Then
-                    Assert.True(False, messages.ToString())
+                    Assert.True(False, .ToString())
                 Else
-                    CompareLineByLine(expectedText, actualText)
+                    Dim expectedLines = expectedText.Split({vbCrLf}, StringSplitOptions.RemoveEmptyEntries)
+                    Dim actualLines = actualText.Split({vbCrLf}, StringSplitOptions.RemoveEmptyEntries)
+
+                    Dim appendedLines As Integer = 0
+
+                    .AppendLine("MISSING ERROR MESSAGES:")
+                    For Each l In expectedLines
+                        If Not actualLines.Contains(l) Then
+                            .AppendLine(l)
+                            appendedLines += 1
+                        End If
+                    Next
+
+                    .AppendLine("UNEXPECTED ERROR MESSAGES:")
+                    For Each l In actualLines
+                        If Not expectedLines.Contains(l) Then
+                            .AppendLine(l)
+                            appendedLines += 1
+                        End If
+                    Next
+
+                    If appendedLines > 0 Then
+                        Assert.True(False, .ToString())
+                    Else
+                        CompareLineByLine(expectedText, actualText)
+                    End If
                 End If
-
-            End If
+            End With
+            messages.Free()
         End If
     End Sub
 
     Private Sub CompareLineByLine(expected As String, actual As String)
         Dim expectedReader = New StringReader(expected)
         Dim actualReader = New StringReader(actual)
 
-        Dim expectedBuilder As New StringBuilder()
-        Dim actualBuilder As New StringBuilder()
+        Dim expectedBuilder = PooledStringBuilderPool.Allocate()
+        Dim actualBuilder = PooledStringBuilderPool.Allocate()
         Dim expectedLine = expectedReader.ReadLine()
         Dim actualLine = actualReader.ReadLine()
 
         While expectedLine IsNot Nothing AndAlso actualLine IsNot Nothing
             If Not expectedLine.Equals(actualLine) Then
-                expectedBuilder.AppendLine("<! " & expectedLine)
-                actualBuilder.AppendLine("!> " & actualLine)
+                expectedBuilder.Builder.AppendLine("<! " & expectedLine)
```

Comment:
>expectedBuilder.Builder [](start = 16, length = 23)

Please extract the ```expectedBuilder.Builder``` and ```actualBuilder.Builder``` into locals and use the locals throughout the function. Ideally, use ```expectedBuilder``` and ```actualBuilder``` as the names for the locals and use different names for PooledStringBuilders, that would decrease an unnecessary churn in this function.